kafl_fuzzer/common/config/settings.py

```diff
@@ -0,0 +1,174 @@
+import os
+import re
+from pathlib import Path
+from argparse import Namespace
+
+from appdirs import AppDirs
+from dynaconf import Dynaconf, Validator
+
+from typing import List, Optional, Any
+
+CUR_DIR = Path(__file__).parent
+APPNAME = 'kAFL'
+SETTINGS_FILENAME = "settings.yaml"
+
+VALID_DEBUG_ACTIONS = ["benchmark", "gdb", "trace", "single", "trace-qemu", "noise", "printk", "redqueen",
+                   "redqueen-qemu", "verify"]
+
+def app_settings_files() -> List[str]:
+    settings_files = [
+        # default config
+        str(CUR_DIR / f"default_{SETTINGS_FILENAME}"),
+        # /etc/xdg/kAFL/
+        str(Path(appdirs.site_config_dir) / SETTINGS_FILENAME),
+        # $HOME/.config/kAFL
+        str(Path(appdirs.user_config_dir) / SETTINGS_FILENAME),
+        # local where the program is started
+        SETTINGS_FILENAME
+    ]
+    # env var KAFL_CONFIG_FILE if present
+    try:
+        settings_files.append(os.environ['KAFL_CONFIG_FILE'])
```

Comment:
If KAFL_CONFIG_FILE is set but points to a non-existing files, this is silently ignored. If the file is inaccessible, it prints an error but continues launching the fuzzer. I think we want to throw an error and stop launching in this case, since it is likely a mistake the user would like to catch?

It also acts a bit funny if the file has bad content: A "test:" token is silently ignored, a "test" line throws some deep exception from dynaconf parsing.